Bottom: f341ad5508523bdb8d223a85d6eba42764f2f589
Top:    2b7e6869699265ab2ad9b2531d9d501c7515e49e
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-27 15:54:21 -0500

Updates to openldap

Wait on openldap container startup

Occasionally the openldap container would not be ready when attempting
to add the admin and proxyagent accounts. A wait is required to
ensure the service is running properly.

---
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 9c102e67..c9f3eccf 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -107,7 +107,7 @@
 # *****************************************************************************
 # Start the data container running
 
-- name: start container (ldap)
+- name: create container (ldap)
   docker_container:
     name: '{{ openldap_container_name }}'
     detach: true
@@ -123,6 +123,7 @@
     image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
     networks:
       - name: '{{ docker_network_frontend }}'
+    state: present
 
 - name: start container (phpldapadmin)
   docker_container:
@@ -146,10 +147,18 @@
 
 - include_tasks: restore.yml
   when: st_openldap_restore.stat.exists == False
+  tags: restore
 
 # *****************************************************************************
 # Set the LDAP passwords
 
+- name: wait for LDAP to start
+  wait_for:
+    delay: 5
+    host: '{{ openldap_ip_addr }}'
+    port: 389
+    state: started
+
 - name: set LDAP admin password
   command: 'docker exec -i {{ openldap_container_name }} /docker-entrypoint.sh set_admin_pass {{ openldap_admin_password | quote }}'
 
diff --git a/roles/docker-openldap/tasks/restore.yml b/roles/docker-openldap/tasks/restore.yml
index d139794e..755e9106 100644
--- a/roles/docker-openldap/tasks/restore.yml
+++ b/roles/docker-openldap/tasks/restore.yml
@@ -31,14 +31,28 @@
 # restore the openldap backup
 
 - name: stop container (ldap)
-  command: 'docker stop {{ openldap_container_name }}'
+  docker_container:
+    name: '{{ openldap_container_name }}'
+    state: stopped
 
-- name: load ldif files
-  command: 'docker run --rm --volume {{ openldap_cv_name }}:/etc/ldap:z --volume {{ openldap_dv_name }}:/var/lib/ldap:z -v {{ host_openldap_docker_restore_dir }}:/tmp/import_export {{ openldap_image_name }}:{{ docker_image_tag }} apply_ldif -n {{ item.dbnum }} -l {{ item.ldif_file }}'
+- name: restore ldif files
+  docker_container:
+    name: '{{ openldap_container_name }}-restore-{{ item.dbnum }}'
+    cleanup: true
+    detach: false
+    volumes:
+      - '{{ openssl_dv_name }}:/etc/ssl/private:z'
+      - '{{ openldap_cv_name }}:/etc/ldap:z'
+      - '{{ openldap_dv_name }}:/var/lib/ldap:z'
+      - '{{ openldap_docker_restore_dir }}:/tmp/import_export:z'
+    image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
+    command: ['apply_ldif', '-n', '{{ item.dbnum }}', '-l', '{{ item.ldif_file }}']
   with_items: '{{ openldap_backup_files }}'
 
 - name: restart container (ldap)
-  command: 'docker start {{ openldap_container_name }}'
+  docker_container:
+    name: '{{ openldap_container_name }}'
+    state: started
 
 # *****************************************************************************
 # cleanup
